src/type.jl

```diff
@@ -13,13 +15,46 @@ Base.getindex(x::Geometry{PointTrait,D,T}, i) where {D,T} = Base.getindex(x.geom
 GeoInterface.isgeometry(::Type{<:Geometry}) = true
 GeoInterface.ncoord(_, ::Geometry{X,D}) where {X,D} = D
 GeoInterface.getcoord(::PointTrait, g::Geometry, i) = Base.getindex(g.geom, i)
+GeoInterface.getcoord(::PointTrait, g::Geometry{X,D,T,<:GeoFormatTypes.MixedFormat}, i) where {X,D,T} = getcoord(PointTrait(), g.geom, i)
 GeoInterface.geomtrait(::Geometry{X}) where {X} = X()
 GeoInterface.ngeom(_, g::Geometry) = length(g.geom)
+GeoInterface.ngeom(t, g::Geometry{X,D,T,<:GeoFormatTypes.MixedFormat}) where {X,D,T} = ngeom(t, g.geom)
 GeoInterface.getgeom(_, g::Geometry, i) = Base.getindex(g, i)
+GeoInterface.getgeom(t, g::Geometry{X,D,T,<:GeoFormatTypes.MixedFormat}, i) where {X,D,T} = getgeom(t, g.geom, i)
 
 childtrait(::LineStringTrait) = PointTrait
 childtrait(::LinearRingTrait) = PointTrait
 childtrait(::PolygonTrait) = LinearRingTrait
 childtrait(::MultiPointTrait) = PointTrait
 childtrait(::MultiLineStringTrait) = LineStringTrait
 childtrait(::MultiPolygonTrait) = PolygonTrait
+
+# For writing
+abstract type AbstractEncoding end
+abstract type AbstractNativeEncoding <: AbstractEncoding end
+struct Seperated <: AbstractNativeEncoding end
+struct Interleaved <: AbstractNativeEncoding end
+struct WellKnownBinary <: AbstractEncoding end
+struct WellKnownText <: AbstractEncoding end
+
+struct Wrapper{E,T,N,G}
+    geom::G
+end
+Base.:(==)(x::Wrapper{E,T,N,G}, y::Wrapper{E,T,N,G}) where {E,T,N,G} = x.geom == y.geom
+Base.show(io::IO, ::Wrapper{E,T,G}) where {E,T,G} = print(io, "$T geometry encoded as $E")
+Wrapper(e::AbstractEncoding, x) = Wrapper{typeof(e),typeof(GeoInterface.geomtrait(x)),GeoInterface.ncoord(x),typeof(x)}(x)
+Wrapper(x) = Wrapper(Interleaved(), x)
+
+data(x::Wrapper{E,T,N,G}) where {E,T,N,G} = _coordinates(E(), T(), N, x.geom)
```

Comment:
```suggestion
data(x::Wrapper{E,T,N,G}) where {E,T,N,G} = _coordinates(E(), T(), Val{N}(), x.geom)
```

Lets not leave this up to the compiler to constant propagate, better to just keep it as a type.